common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -260,67 +260,69 @@ public static CharSequence escapeCsv(CharSequence value) {
      */
     public static CharSequence escapeCsv(CharSequence value, boolean trimWhiteSpace) {
         int length = checkNotNull(value, "value").length();
-        if (length == 0) {
-            return value;
-        }
-
-        int start = 0;
-        int last = length - 1;
-        boolean trimmed = false;
+        int start;
+        int last;
         if (trimWhiteSpace) {
             start = indexOfFirstNonOwsChar(value, length);
-            if (start == length) {
-                return EMPTY_STRING;
-            }
             last = indexOfLastNonOwsChar(value, start, length);
-            trimmed = start > 0 || last < length - 1;
-            if (trimmed) {
-                length = last - start + 1;
+        } else {
+            start = 0;
+            last = length - 1;
+        }
+        if (start > last) {
+            return EMPTY_STRING;
+        }
+
+        int firstUnescapedSpecial = -1;
+        boolean quoted = false;
+        if (isDoubleQuote(value.charAt(start))) {
+            quoted = isDoubleQuote(value.charAt(last)) && last > start;
+            if (quoted) {
+                start++;
+                last--;
+            } else {
+                firstUnescapedSpecial = start;
             }
         }
 
-        StringBuilder result = new StringBuilder(length + CSV_NUMBER_ESCAPE_CHARACTERS);
-        boolean quoted = isDoubleQuote(value.charAt(start)) && isDoubleQuote(value.charAt(last)) && length != 1;
-        boolean foundSpecialCharacter = false;
-        boolean escapedDoubleQuote = false;
-        for (int i = start; i <= last; i++) {
-            char current = value.charAt(i);
-            switch (current) {
-                case DOUBLE_QUOTE:
-                    if (i == start || i == last) {
-                        if (!quoted) {
-                            result.append(DOUBLE_QUOTE);
-                        } else {
-                            continue;
-                        }
-                    } else {
-                        boolean isNextCharDoubleQuote = isDoubleQuote(value.charAt(i + 1));
-                        if (!isDoubleQuote(value.charAt(i - 1)) &&
-                                (!isNextCharDoubleQuote || i + 1 == last)) {
-                            result.append(DOUBLE_QUOTE);
-                            escapedDoubleQuote = true;
-                        }
+        if (firstUnescapedSpecial < 0) {
+            for (int i = start; i <= last; i++) {
+                char c = value.charAt(i);
+                if (!quoted && (c == LINE_FEED || c == CARRIAGE_RETURN || c == COMMA)) {
```

Comment:
have you considered unrolling this loop? `quoted` will not change during this loop.